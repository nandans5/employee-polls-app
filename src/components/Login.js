import { useState } from "react";
import { setAuthedUser } from "../actions/authedUser";
import { connect } from "react-redux";

const Login = ({ users, dispatch }) => {
  const [userid, setUserid] = useState("");
  const [password, setPassword] = useState("");

  const handleSubmit = (e) => {
    e.preventDefault();

    if (users[userid] && users[userid].password === password) {
      dispatch(setAuthedUser(userid));
    } else {
      alert("login failed");
    }

    setUserid("");
    setPassword("");
  };

  const handleChangeUserid = (e) => {
    const text = e.target.value;

    setUserid(text);
  };

  const handleChangePassword = (e) => {
    const text = e.target.value;

    setPassword(text);
  };

  return (
    <div>
      <h2 className="center">Login Page</h2>
      <form className="new-tweet" onSubmit={handleSubmit}>
        <label>User ID</label>
        <textarea
          placeholder="enter userid"
          value={userid}
          onChange={handleChangeUserid}
          className="text-area"
        />
        <label>Password</label>
        <textarea
          placeholder="enter password"
          value={password}
          onChange={handleChangePassword}
          className="text-area"
        />
        <button
          className="btn"
          type="submit"
          disabled={userid === "" || password === ""}
        >
          Submit
        </button>
      </form>
    </div>
  );
};

const mapStateToProps = ({ users, dispatch }) => {
  return {
    users,
    dispatch,
  };
};

export default connect(mapStateToProps)(Login);
