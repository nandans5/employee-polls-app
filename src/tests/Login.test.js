import { fireEvent, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import React from "react";
import Login from "../components/Login";
import { store } from "../store";

describe("Login", () => {
  it("match screenshot", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  it("verify ids", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );

    const userid = component.getByTestId("userid-input");
    const password = component.getByTestId("password-input");
    const submitbtn = component.getByTestId("submit-btn");

    expect(userid).toBeInTheDocument();
    expect(password).toBeInTheDocument();
    expect(submitbtn).toBeInTheDocument();
  });
  it("check fireevent", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );

    const userid = component.getByTestId("userid-input");
    const password = component.getByTestId("password-input");
    const submitbtn = component.getByTestId("submit-btn");

    expect(submitbtn).toBeDisabled();

    fireEvent.change(userid, { target: { value: "sarahedo" } });
    fireEvent.change(password, { target: { value: "password" } });

    expect(submitbtn).toBeEnabled();
  });
});
