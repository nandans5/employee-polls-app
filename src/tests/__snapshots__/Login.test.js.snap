// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login match screenshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2
          class="center"
        >
          Login Page
        </h2>
        <form
          class="login-form"
        >
          <label>
            User ID
          </label>
          <textarea
            class="text-area"
            data-testid="userid-input"
            placeholder="enter userid"
          />
          <label>
            Password
          </label>
          <textarea
            class="text-area"
            data-testid="password-input"
            placeholder="enter password"
          />
          <button
            class="btn"
            data-testid="submit-btn"
            disabled=""
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        class="center"
      >
        Login Page
      </h2>
      <form
        class="login-form"
      >
        <label>
          User ID
        </label>
        <textarea
          class="text-area"
          data-testid="userid-input"
          placeholder="enter userid"
        />
        <label>
          Password
        </label>
        <textarea
          class="text-area"
          data-testid="password-input"
          placeholder="enter password"
        />
        <button
          class="btn"
          data-testid="submit-btn"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
